### DISCUSSION TITLE
Stock movement reports

### DISCUSSION THREAD

**Discussion by snorkrat** [inventree/InvenTree#8152] [Q&A]
Is there a way to generate a report/spreadsheet of stock movements, ideally specifying a date range?

I would like to select a stock item, specify a start date and end date and see the stock qty at the start, all stock movements (sales orders, adjustments, etc), and then the final stock at the end date.

Thanks for the help!

**Comment by SchrodingersGat** [comment]
In the new interface you can simply download the stock history data from the stock item detail page:

![image](https://github.com/user-attachments/assets/312416bb-a763-4cdf-b28b-cc80b62315e3)

**Reply by NotADeveloper-netizen** [reply]
Is there anyway to view stock history or stock movement on the 	0.17.0 dev version?

**Reply by SchrodingersGat** [reply]
Yes, go to the stock item page, and click on the "stock tracking" tab

**Comment by 78kfb** [comment]
Is there a way to view the stock movement history for all inventory items over a selected time period, rather than for a single part? Are there any built-in analytics or reports that provide an overview of warehouse activity and inventory movements?
